Lib/shutil.py

```diff
@@ -813,6 +813,10 @@ def move(src, dst, copy_function=copy2):
             if _destinsrc(src, dst):
                 raise Error("Cannot move a directory '%s' into itself"
                             " '%s'." % (src, dst))
+            if not os.access(src, os.W_OK) and os.listdir(src):
```

Comment:
Please add a test at https://github.com/python/cpython/blob/master/Lib/test/test_shutil.py to cover this scenario.

Reply:
Ack, see my comment below.